modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -799,14 +799,22 @@ else if (navigation.equals("mine") || navigation.equals("recent")) {
 	private Hits _getHits(SearchContainer<RepositoryEntry> searchContainer)
 		throws PortalException {
 
+		long searchRepositoryId = ParamUtil.getLong(
+			_httpServletRequest, "searchRepositoryId");
+
+		if (searchRepositoryId != 0) {
+			ThemeDisplay themeDisplay =
```

Comment:
You are modifying the themeDisplay only use the value on the search context

why do not do something like :
`searchContext.setGroupIds(new long[] {searchRepositoryId});`


It's only an idea/suggestion

Reply:
I tried it before. I think both solution are the same. I decided choose updating themeDisplay after searching "themeDisplay.setScopeGroupId" keyword in liferay-portal source. It is quite commonly used